src/mappings/spacewalk.ts

```diff
@@ -0,0 +1,184 @@
+
+    await ctx.store.save(issueRequest)
+}
+
+export async function handleIssueRequestAmountChanged(
+    ctx: EventHandlerContext
+) {
+    const { args } = ctx.event
+    const { issueId, confiscatedGriefingCollateral } = args
+
+    let issueRequest = await ctx.store.get(IssueRequest, issueId)
+
+    if (issueRequest == null) {
+        throw new Error(
+            'Issue request MUST exists on the database when handling cancel event'
+        )
+    }
+
+    // we update on this event the slashed collateral if the amount sent was not enough. The amount is updated on the executed event.
+    issueRequest.slashedCollateral = confiscatedGriefingCollateral
+    await ctx.store.save(issueRequest)
+}
+
+export async function handleRedeemRequest(ctx: EventHandlerContext) {
+    const { args } = ctx.event
+    const { redeemId, redeemer, amount, vaultId, fee, premium, transferFee } =
+        args
+    const vaultIdFlat = parseVaultId(vaultId)
+    const vault = await getOrCreateVault(
+        ctx,
+        vaultIdFlat,
+        undefined // It doesn't matter we don't have this, the vault should have already issued something and therefore must have been created.
+    )
+
+    const redeemRequestStorage = await getVersionedStorage(
+        network,
+        ctx,
+        'redeem',
+        'redeemRequests'
+    )
+    const createdRedeem = await redeemRequestStorage.get(ctx.block, redeemId)
+
+    const redeemRequest = new RedeemRequest({
+        id: redeemId,
+        timestamp: new Date(ctx.block.timestamp ?? 0),
+        opentime: createdRedeem.opentimes,
+        period: createdRedeem.period,
+        redeemer: hexToSs58(redeemer),
+        amount: amount,
+        vault: vault,
+        fee: fee,
+        premium: premium,
+        transferFee: transferFee,
+        status: RedeemRequestStatus.PENDING,
+    })
+
+    await ctx.store.save(redeemRequest)
+}
+
+export async function handleRedeemRequestExecuted(ctx: EventHandlerContext) {
+    const { args } = ctx.event
+    const { redeemId } = args
+
+    let redeemRequest = await ctx.store.get(RedeemRequest, redeemId)
+
+    if (redeemRequest == null) {
+        throw new Error(
+            'Redeem request MUST exists on the database when handling executed event'
+        )
+    }
+
+    redeemRequest.status = RedeemRequestStatus.COMPLETED
+
+    await ctx.store.save(redeemRequest)
+}
+
+export async function handleRedeemRequestCancelled(ctx: EventHandlerContext) {
+    const { args } = ctx.event
+    const { redeemId, status, slashedAmount } = args
+
+    let redeemRequest = await ctx.store.get(RedeemRequest, redeemId)
+
+    if (redeemRequest == null) {
+        throw new Error(
+            'Redeem request MUST exists on the database when handling cancel event'
+        )
+    }
+
+    if (status == 'Reimbursed') {
+        redeemRequest.status = RedeemRequestStatus.REIMBURSED
+        // Todo, not sure how this status is represented.
+    } else if (status == 'Reimbursed(true)?') {
```

Comment:
Is the stringified status really `Reimbursed` or `Reimbursed(true)?` (with the question mark) when the Rust enum defines it as `Reimbursed(bool)`?